lib/index.js

```diff
@@ -0,0 +1,146 @@
+'use strict'
+const Promise = require('bluebird')
+const yaml = require('js-yaml')
+const Joi = require('joi')
+const hostname = require('@runnable/hostname')
+
+const removeNullAndUndefinedKeys = require('./util').removeNullAndUndefinedKeys
+const validate = require('./util').validate
+const Parser = require('./parser')
+const Warning = require('./warning')
+
+const dockerComposeFileSchema = Joi.object({
+  version: Joi.string().valid('2').required(),
+  services: Joi.object().required()
+}).unknown().required()
+
+const arrayOfStrings = Joi.array().items(Joi.string())
+
+const serviceSchemaKeys = {
+  build: Joi.alternatives().try(Joi.string(), Joi.object()),
+  image: Joi.string(), // TODO: Add parser
+  command: Joi.alternatives().try(Joi.string(), arrayOfStrings),
+  ports: arrayOfStrings,
+  expose: arrayOfStrings, // TODO
+  links: arrayOfStrings, // TODO:
+  depends_on: arrayOfStrings, // TODO
+  environment: arrayOfStrings
+}
+const serviceSchema = Joi.object(serviceSchemaKeys).unknown().required()
+
+module.exports = class DockerComposeParser {
+
+  static parse (opts) {
```

Comment:
Because this to be supported in node 4 without compiling down! If we used v6 in API I might have just done it for v6. Node 4 doesn't support those features.